device/celestica/x86_64-cel_silverstone-r0/plugins/sfputil.py

```diff
@@ -344,11 +439,95 @@ def reset(self, port_num):
+        # Workaround script to bring up port link
+        for port_num in int_sfp:
+
+            if int_sfp[port_num] == '1':
+                i2c_num = int(port_num) + self.EEPROM_OFFSET
+
+                # run cmis init script
+                init_cmd = "bash {} {}".format(init_script, i2c_num)
+                self._run_command(init_cmd)
+
+    def _get_platform_and_hwsku(self):
+        try:
+            proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-H', '-v', PLATFORM_KEY],
+                                    stdout=subprocess.PIPE,
+                                    shell=False,
+                                    stderr=subprocess.STDOUT)
+            stdout = proc.communicate()[0]
+            proc.wait()
+            platform = stdout.rstrip('\n')
+
+            proc = subprocess.Popen([SONIC_CFGGEN_PATH, '-d', '-v', HWSKU_KEY],
+                                    stdout=subprocess.PIPE,
+                                    shell=False,
+                                    stderr=subprocess.STDOUT)
+            stdout = proc.communicate()[0]
+            proc.wait()
+            hwsku = stdout.rstrip('\n')
+        except OSError, e:
+            raise OSError("Cannot detect platform")
+
+        return (platform, hwsku)
+
+    def get_path_to_cmis_init_file(self):
+        (platform, hwsku) = self._get_platform_and_hwsku()
+        platform_path = "/".join([PLATFORM_ROOT_PATH, platform])
+        hwsku_path = "/".join([platform_path, hwsku])
+        port_config_file_path = "/".join([hwsku_path, "cmis-init.sh"])
+        return port_config_file_path
+
     def get_transceiver_change_event(self, timeout=0):
         """
-        TBD
+        :param timeout in milliseconds. The method is a blocking call. When timeout is 
+         zero, it only returns when there is change event, i.e., transceiver plug-in/out
+         event. When timeout is non-zero, the function can also return when the timer expires.
+         When timer expires, the return status is True and events is empty.
+        :returns: (status, events)
+        :status: Boolean, True if call successful and no system level event/error occurred, 
+         False if call not success or system level event/error occurred.
+        :events: dictionary for physical port index and the SFP status,
+         status='1' represent plug in, '0' represent plug out like {'0': '1', '31':'0'}
+         when it comes to system level event/error, the index will be '-1',
+         and status can be 'system_not_ready', 'system_become_ready', 'system_fail',
+         like {'-1':'system_not_ready'}.
         """
-        raise NotImplementedError
+        cmis_init_script = self.get_path_to_cmis_init_file()        
+        sfp_event = SfpEvent(self.NUM_OSFP, self.PORT_INFO_PATH)
+        start_milli_time = int(round(time.time() * 1000))
+        int_sfp = {}
+
+        sleep_time = min(
+            timeout, self.POLL_INTERVAL) if timeout != 0 else self.POLL_INTERVAL
```

Comment:
timeout is milliseconds, what is POLL_INTERVAL unit? If it is seconds, this min() parameters do not seem to be correct.